### ISSUE TITLE
Update to logicalcluster v2

### ISSUE THREAD

**Issue by ncdc** [kcp-dev/controller-runtime#18]

**Comment by pradeepnnv** [comment]
Hi @ncdc , this involves updating go.mod ?

github.com/kcp-dev/logicalcluster v2.0.0-alpha.0

I wasn't sure if I should use an alpha version here.

**Comment by ncdc** [comment]
Hi @pradeepnnv, yes, that's what is needed for now. We'll have a newer v2.x tag of logicalcluster in the future, but for the time being, that's the one to use.

**Comment by pradeepnnv** [comment]
Hi @ncdc , 

I updated the go.mod as below.

```
diff --git a/go.mod b/go.mod
index 3823a930..6dc27c10 100644
--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
        github.com/go-logr/logr v1.2.0
        github.com/go-logr/zapr v1.2.0
        github.com/kcp-dev/apimachinery v0.0.0-20220708220956-c302aeddfde7
-       github.com/kcp-dev/logicalcluster v1.1.1-0.20220705215104-8e46328c24a5
+       github.com/kcp-dev/logicalcluster/v2 v2.0.0-alpha.0
        github.com/onsi/ginkgo v1.16.5
        github.com/onsi/gomega v1.17.0
        github.com/prometheus/client_golang v1.11.0
```

But, most of the tests are failing when I run `make test`. 
Please let me know if I'm doing anything wrong.

**Comment by ncdc** [comment]
There's some work that needs to be done to get the tests functional in our fork. Feel free to open a PR and not worry about the tests for now (while we get them working again).

**Comment by stevekuznetsov** [comment]
Fixed in #21 
/close
